src/android/java/com/pspdfkit/cordova/action/xfdf/ImportXfdfAction.java

```diff
@@ -0,0 +1,78 @@
+package com.pspdfkit.cordova.action.xfdf;
+
+import android.net.Uri;
+
+import androidx.annotation.NonNull;
+
+import com.pspdfkit.annotations.Annotation;
+import com.pspdfkit.cordova.CordovaPdfActivity;
+import com.pspdfkit.cordova.PSPDFKitCordovaPlugin;
+import com.pspdfkit.cordova.action.BasicAction;
+import com.pspdfkit.document.PdfDocument;
+import com.pspdfkit.document.formatters.XfdfFormatter;
+import com.pspdfkit.document.providers.ContentResolverDataProvider;
+import com.pspdfkit.ui.PdfFragment;
+
+import org.apache.cordova.CallbackContext;
+import org.apache.cordova.PluginResult;
+import org.json.JSONArray;
+import org.json.JSONException;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+/**
+ * Asynchronously imports annotations from XFDF-file to the current document.
+ */
+public class ImportXfdfAction extends BasicAction {
+
+  private static final int ARG_XFDF_FILE_URI = 0;
+
+  public ImportXfdfAction(@NonNull String name, @NonNull PSPDFKitCordovaPlugin plugin) {
+    super(name, plugin);
+  }
+
+  @Override
+  protected void execAction(JSONArray args, CallbackContext callbackContext) throws JSONException {
+    final Uri xfdfFileUri = Uri.parse(args.getString(ARG_XFDF_FILE_URI));
```

Comment:
Same question about string vs. file access here. Our JSON API actually takes the JSON string instead of a file.

Reply:
Similar to iOS. I imagine it can be modified to support just a string if needed be. I am not exactly sure what caused this API decision on iOS TBH. @radazzouz could you comment on why we decided to use files instead of formatted strings?

Reply:
Good question! On Instant JSON we work with JSON payloads. The XFDF format itself is based on files. People who work with it are used to have XFDF files. Also, the iOS APIs are tailored to use files rather than XML payloads.